app/src/main/java/pl/gittobefit/WorkoutDisplay/fragments/DisplayReceivedTraining.java

```diff
@@ -51,14 +67,47 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         model.getPosition().observe(getViewLifecycleOwner(), integer -> index = integer);
     }
 
+
     @Override
     public void onResume() {
 
         super.onResume();
+        List<SavedTraining> result2 = AppDataBase.getInstance(getContext()).training().getInfoForTrainingList();
+
+        if (index != UserTrainings.getInstance().getTrainingArrayList().size() && UserTrainings.getInstance().getTraining(index).getPlanList().size() == 0) {
+            SavedTraining savedTraining = AppDataBase.getInstance(getContext()).training().getOneTraining(result2.get(index).getId());
+            TrainingPlan trainingPlan;
+            Training training = new Training();
+            Exercise exercise;
+            ArrayList<ExerciseExecution> exerciseExecutionArrayList;
+            ArrayList<TrainingPlan> trainingPlanArrayList;
+            trainingPlanArrayList = new ArrayList<>();
+            for (int k = 0; k < savedTraining.getPlanList().size(); k++) {
+                exerciseExecutionArrayList = new ArrayList<>();
+                for (int j = 0; j < savedTraining.getPlanList().get(k).size(); j++) {
+                    ExerciseExecution exerciseExecution = new ExerciseExecution();
+                    exercise = new Exercise();
+                    exercise.setName(AppDataBase.getInstance(getContext()).exercise().getExerciseList(savedTraining.getPlanList().get(k).get(j).getExerciseId()).getName());
+                    exerciseExecution.setExercise(exercise);
+                    exerciseExecution.setCount(savedTraining.getPlanList().get(k).get(j).getCount());
+                    exerciseExecution.setSeries(savedTraining.getPlanList().get(k).get(j).getSeries());
+                    exerciseExecution.setTime(savedTraining.getPlanList().get(k).get(j).getTime());
+                    exerciseExecutionArrayList.add(exerciseExecution);
+                }
+                trainingPlan = new TrainingPlan();
+                trainingPlan.setExercisesExecutions(exerciseExecutionArrayList);
+
+                trainingPlanArrayList.add(trainingPlan);
+                training.setPlanList(trainingPlanArrayList);
+                UserTrainings.getInstance().getTraining(index).setPlanList(trainingPlanArrayList);
+            }
+        }
```

Comment:
Wydaje mi się, że jakbyś korzystał z repozytorium wystawionego przez Dominika, to nie trzeba by było ręcznie mapować. (Jeżeli nie będziesz chciał z niego korzystać, to może lepiej jest to wydzielić do metody prywatnej)